### PR TITLE
Support autocomplete tags in traces query (backend side), Replace all configurations **_JETTY_** to **_REST_**

### PR DESCRIPTION
**Pull request by wankai123**
- [X] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [X] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/changelog/docs/en/changes/changes.md).

related to https://github.com/apache/skywalking/issues/8933, the Trace part.

update:
[Breaking Change] Replace all configurations `**_JETTY_**` to `**_REST_**`.

### CODE REVIEW

File: oap-server/server-starter/src/main/resources/application.yml [hunk 1]
```diff
@@ -393,6 +393,7 @@ receiver-zipkin:
     jettyAcceptorPriorityDelta: ${SW_RECEIVER_ZIPKIN_JETTY_DELTA:0}
     jettyAcceptQueueSize: ${SW_RECEIVER_ZIPKIN_QUEUE_SIZE:0}
     instanceNameRule: ${SW_RECEIVER_ZIPKIN_INSTANCE_NAME_RULE:[spring.instance_id,node_id]}
+    searchableTracesTags: ${SW_ZIPKIN_SEARCHABLE_TAG_KEYS:http.method}
```
Comment (kezhenxu94) [inline oap-server/server-starter/src/main/resources/application.yml:396]: Add this to the doc please

File: docs/en/changes/changes.md [hunk 2]
```diff
@@ -30,6 +30,7 @@
 * Add data-generator module to run OAP in testing mode, generating mock data for testing.
 * Support receive Kubernetes processes from gRPC protocol.
 * Fix the problem that es index(TimeSeriesTable, eg. endpoint_traffic, alarm_record) didn't create even after rerun with init-mode. This problem caused the OAP server to fail to start when the OAP server was down for more than a day.
+* Support autocomplete tags in traces query (backend side).
```
Comment (wu-sheng) [inline docs/en/changes/changes.md:33]:
```suggestion
* Support autocomplete tags in traces query.
```

This change log is in the OAP side already.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/searchtag/TagAutocompleteData.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Override
+    public Metrics toDay() {
+        return null;
+    }
+
+    @Override
+    protected String id0() {
+        return toTimeBucketInDay() + "-" + tag;
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/searchtag/TagAutocompleteData.java:69]: The tag is not persistent, when this is deserialized from storage, this tag is `null`.

Are you counting on this method is never called because `supportUpdate = false` in the sub classes? I think this may be true for now, but not a good practice, and hard to follow besides we two.

**Comment by wu-sheng** [comment]
@dmsolr Please check InfluxDB relative parts. @LIU-WEI-git Please check IoTDB relative parts.

File: oap-server/server-starter/src/main/resources/application.yml (hunk 1 above)
Comment (wankai123) [reply oap-server/server-starter/src/main/resources/application.yml:395]: I have added this and some missing items to the doc.  I found the env. variable still name `JETTY_xx`, should we keep these name styles for compatibility?

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/searchtag/TagAutocompleteData.java (hunk 3 above)
Comment (wankai123) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/searchtag/TagAutocompleteData.java:69]: Done, set in `storage2Entity`

**Review by wu-sheng — DISMISSED** [review]

File: docs/en/setup/backend/configuration-vocabulary.md [hunk 4]
```diff
@@ -202,10 +202,16 @@ core|default|role|Option values: `Mixed/Receiver/Aggregator`. **Receiver** mode
...
+| - | - | port | Binding port of RESTful services. | SW_RECEIVER_ZIPKIN_PORT| 9411                                                                  |
+| - | - | contextPath| Web context path of RESTful services. | SW_RECEIVER_ZIPKIN_CONTEXT_PATH| /                                                                     |
+| - | - | maxThreads| Maximum thread number of RESTful services. | SW_RECEIVER_ZIPKIN_JETTY_MAX_THREADS| 200                                                                   |
+| - | - | idleTimeOut| Connector idle timeout of RESTful services (in milliseconds). | SW_RECEIVER_ZIPKIN_JETTY_IDLE_TIMEOUT| 30000                                                                 |
+| - | - | acceptorPriorityDelta| ServerSocketChannel backlog of RESTful services. | SW_RECEIVER_ZIPKIN_JETTY_DELTA| 0                                                                     |
+| - | - | acceptQueueSize| Maximum request header size accepted. | SW_RECEIVER_ZIPKIN_QUEUE_SIZE| 0                                                                     |
```
Comment (wu-sheng) [inline docs/en/setup/backend/configuration-vocabulary.md:212]: Let's remove Jetty from all places. It is in v9 iteration now, we don't need this debt

Comment (wu-sheng) [reply docs/en/setup/backend/configuration-vocabulary.md:212]: `All` includes `jetty*` in core module and all other places. These should be changed in 9.0 release, let's do this sooner than later.

File: oap-server/server-starter/src/main/resources/application.yml (hunk 1 above)
Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/application.yml:395]: See https://github.com/apache/skywalking/pull/8971#discussion_r861007722

File: docs/en/setup/backend/configuration-vocabulary.md (hunk 4 above)
Comment (wankai123) [reply docs/en/setup/backend/configuration-vocabulary.md]: Removed all, @kezhenxu94 pls check.

**Review by wu-sheng — DISMISSED** [review]

**Review by wu-sheng — APPROVED** [review]

File: oap-server/server-starter/src/main/resources/application.yml [hunk 5]
```diff
@@ -384,15 +382,16 @@ receiver-otel:
...
-    jettyIdleTimeOut: ${SW_RECEIVER_ZIPKIN_JETTY_IDLE_TIMEOUT:30000}
-    jettyAcceptorPriorityDelta: ${SW_RECEIVER_ZIPKIN_JETTY_DELTA:0}
-    jettyAcceptQueueSize: ${SW_RECEIVER_ZIPKIN_QUEUE_SIZE:0}
+    # For HTTP server
+    restHost: ${SW_RECEIVER_ZIPKIN_REST_HOST:0.0.0.0}
+    restPort: ${SW_RECEIVER_ZIPKIN_REST_PORT:9411}
+    restContextPath: ${SW_RECEIVER_ZIPKIN_REST_CONTEXT_PATH:/}
+    restMaxThreads: ${SW_RECEIVER_ZIPKIN_REST_MAX_THREADS:200}
+    restIdleTimeOut: ${SW_RECEIVER_ZIPKIN_REST_IDLE_TIMEOUT:30000}
+    restAcceptorPriorityDelta: ${SW_RECEIVER_ZIPKIN_REST_DELTA:0}
```
Comment (Superskyyy) [inline oap-server/server-starter/src/main/resources/application.yml:391]: I think this configuration is removed already?

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/application.yml:391]: I just merged this a minute ago. Please submit a pull request to fix.

Comment (wankai123) [reply oap-server/server-starter/src/main/resources/application.yml:391]: https://github.com/apache/skywalking/blob/12ba7aa5336af9b323bb2acbc1d864b6b3c416df/oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverProvider.java#L72

not removed for now in zipkin

Comment (Superskyyy) [reply oap-server/server-starter/src/main/resources/application.yml:391]: > https://github.com/apache/skywalking/blob/12ba7aa5336af9b323bb2acbc1d864b6b3c416df/oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverProvider.java#L72
> 
> not removed for now in zipkin

I pointed at the wrong line, I meant the `AcceptorPriorityDelta` config.

Comment (wankai123) [reply oap-server/server-starter/src/main/resources/application.yml:391]: `AcceptorPriorityDelta ` exists, in line 70, I pointed at the wrong line...

Comment (Superskyyy) [reply oap-server/server-starter/src/main/resources/application.yml:391]: > `AcceptorPriorityDelta ` exists, in line 70, I pointed at the wrong line...

Yeah, I mean the config is no longer passed in the actual Armeria server initialization anymore as I can see.

Comment (kezhenxu94) [reply oap-server/server-starter/src/main/resources/application.yml:391]: @Superskyyy can you remove `AcceptorPriorityDelta` in config file and doc? It's not used anymore in Armeria.

Comment (wankai123) [reply oap-server/server-starter/src/main/resources/application.yml:391]: Oh, I see thanks @Superskyyy , you can remove it.

Comment (Superskyyy) [reply oap-server/server-starter/src/main/resources/application.yml:391]: Sure I will remove it now. haha so confusing the code reference mechanism in GitHub..
